subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -533,22 +545,21 @@ For instance an `EntityProcessor` could create a `<TypeName>Repository` for each
 </sample>
 ++++
 
-
 Isolating processors have the following limitations:
 
-* Can't read resources
-* Can't write resources
-* Can't have any side effects except for using the `Filer` and `Messager` APIs
-* Can't depend on compiler-specific APIs like `com.sun.source.util.Trees`
-* Must provide exactly one originating element for each file generated with the `Filer` API
-* Must make all decisions about an element based on information reachable from its AST.
-For instance it can query the super class, method return types etc, but can't look at other, unrelated elements.
+* Must make all decisions (code generation, validation messages) for an annotated type based on information reachable from its AST.
+This means you can analyze the types' super-class, method return types, annotations etc., even transitively.
+But you cannot make decisions based on unrelated elements in the RoundEnvironment.
+Doing so will result in silent failures because too few files will be recompiled later.
+If your processor needs to make decisions based on a combination of otherwise unrelated elements, mark it as 'aggregating' instead.
+* Must provide exactly one originating element for each file generated with the `Filer` API. Providing any other number of elements will result in full recompilation.
+
```

Comment:
As above, the lead-in sentence for the list makes it more natural to prefix each line with "They ..."